code/game/objects/effects/spiders.dm

```diff
@@ -1,213 +1,193 @@
+						forceMove(entry_vent)
+						entry_vent = null
+						return
+
+					if(prob(50))
+						audible_message("<span class='italics'>You hear something scampering through the ventilation ducts.</span>")
+					sleep(travel_time)
+
+					if(!exit_vent || exit_vent.welded)
+						forceMove(entry_vent)
+						entry_vent = null
+						return
+					forceMove(exit_vent.loc)
+					entry_vent = null
+					var/area/new_area = get_area(loc)
+					if(new_area)
+						new_area.Entered(src)
 
-	else if(prob(25))
-		var/list/nearby = oview(5, src)
+	else if(prob(33))
+		var/list/nearby = oview(10, src)
 		if(nearby.len)
 			var/target_atom = pick(nearby)
-			walk_to(src, target_atom, 5)
-			if(prob(25))
-				src.visible_message("<span class='notice'> \the [src] skitters[pick(" away"," around","")].</span>")
-	else if(prob(5))
-		//vent crawl!
+			walk_to(src, target_atom)
+			if(prob(40))
+				visible_message("<span class='notice'>\The [src] skitters[pick(" away"," around","")].</span>")
+	else if(prob(10))
 		for(var/obj/machinery/atmospherics/components/unary/vent_pump/v in view(7,src))
 			if(!v.welded)
 				entry_vent = v
-				walk_to(src, entry_vent, 5)
+				walk_to(src, entry_vent, 1)
 				break
-
-	if(prob(1))
-		src.visible_message("<span class='notice'> \the [src] chitters.</span>")
-	if(isturf(loc) && amount_grown > 0)
+	if(isturf(loc))
 		amount_grown += rand(0,2)
 		if(amount_grown >= 100)
-			var/spawn_type = pick(typesof(/mob/living/simple_animal/hostile/giant_spider))
-			new spawn_type(src.loc)
+			if(!grow_as)
+				if(prob(3))
+					grow_as = pick(/mob/living/simple_animal/hostile/poison/giant_spider/tarantula, /mob/living/simple_animal/hostile/poison/giant_spider/hunter/viper, /mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife)
+				else
+					grow_as = pick(/mob/living/simple_animal/hostile/poison/giant_spider, /mob/living/simple_animal/hostile/poison/giant_spider/hunter, /mob/living/simple_animal/hostile/poison/giant_spider/nurse)
+			var/mob/living/simple_animal/hostile/poison/giant_spider/S = new grow_as(loc)
+			S.faction = faction.Copy()
 			qdel(src)
 
-/obj/effect/decal/cleanable/spiderling_remains
-	name = "spiderling remains"
-	desc = "Green squishy mess."
-	icon = 'icons/effects/effects.dmi'
-	icon_state = "greenshatter"
 
-/obj/effect/spider/cocoon
+/obj/structure/spider/cocoon
 	name = "cocoon"
-	desc = "Something wrapped in silky spider web"
+	desc = "Something wrapped in silky spider web."
 	icon_state = "cocoon1"
 	max_integrity = 60
 
-	New()
-		icon_state = pick("cocoon1","cocoon2","cocoon3")
 
-/obj/effect/spider/cocoon/Destroy()
-	visible_message("<span class='warning'> [src] splits open.</span>")
+/obj/structure/spider/cocoon/Initialize()
+	icon_state = pick("cocoon1", "cocoon2", "cocoon3")
+	return ..()
+
+
+/obj/structure/spider/cocoon/Destroy()
+	var/turf/T = get_turf(src)
+	src.visible_message("<span class='warning'>\The [src] splits open.</span>")
 	for(var/atom/movable/A in contents)
```

Comment:
pretty sure theres a helper proc for this

Reply:
The closes I saw was on_mob_delete, do you know of one for atoms?